Modules/Relationships/src/Relationships.Application/DomainEvents/Incoming/FeatureFlagsOfIdentityChanged/FeatureFlagsOfIdentityChangedDomainEventHandler.cs

```diff
@@ -8,24 +9,58 @@ namespace Backbone.Modules.Relationships.Application.DomainEvents.Incoming.Featu
 public class FeatureFlagsOfIdentityChangedDomainEventHandler : IDomainEventHandler<FeatureFlagsOfIdentityChangedDomainEvent>
 {
     private readonly IRelationshipTemplatesRepository _relationshipTemplatesRepository;
+    private readonly IRelationshipsRepository _relationshipsRepository;
     private readonly IEventBus _eventBus;
 
-    public FeatureFlagsOfIdentityChangedDomainEventHandler(IRelationshipTemplatesRepository relationshipTemplatesRepository, IEventBus eventBus)
+    public FeatureFlagsOfIdentityChangedDomainEventHandler(IRelationshipTemplatesRepository relationshipTemplatesRepository, IEventBus eventBus, IRelationshipsRepository relationshipsRepository)
     {
         _relationshipTemplatesRepository = relationshipTemplatesRepository;
         _eventBus = eventBus;
+        _relationshipsRepository = relationshipsRepository;
     }
 
     public async Task Handle(FeatureFlagsOfIdentityChangedDomainEvent @event)
     {
-        var allocations = await _relationshipTemplatesRepository.FindRelationshipTemplateAllocations(a => a.RelationshipTemplate.CreatedBy == @event.IdentityAddress, CancellationToken.None);
+        var identitiesToBeNotified = await FindAllIdentitiesToBeNotified(@event);
 
-        var publishEventTasks = allocations.Select(allocation => _eventBus.Publish(new PeerFeatureFlagsChangedDomainEvent
+        var publishEventTasks = identitiesToBeNotified.Select(notifiedAddress => _eventBus.Publish(new PeerFeatureFlagsChangedDomainEvent
         {
             PeerAddress = @event.IdentityAddress,
-            NotifiedIdentityAddress = allocation.AllocatedBy
+            NotifiedIdentityAddress = notifiedAddress
         }));
 
         await Task.WhenAll(publishEventTasks);
     }
+
+    private async Task<HashSet<string>> FindAllIdentitiesToBeNotified(FeatureFlagsOfIdentityChangedDomainEvent @event)
+    {
+        var identitiesToBeNotified = new HashSet<string>();
+
+        var activeAndPendingRelationships = await
+            _relationshipsRepository.FindRelationships(
+                r => (r.From == @event.IdentityAddress || r.To == @event.IdentityAddress)
+                     && (r.Status == RelationshipStatus.Active || r.Status == RelationshipStatus.Pending),
+                CancellationToken.None);
+
+        foreach (var relationship in activeAndPendingRelationships)
+        {
+            if (relationship.From == @event.IdentityAddress)
+            {
+                identitiesToBeNotified.Add(relationship.To);
+            }
+            else
+            {
+                identitiesToBeNotified.Add(relationship.From);
+            }
+        }
+
+        var allocations = await _relationshipTemplatesRepository.FindRelationshipTemplateAllocations(a => a.RelationshipTemplate.CreatedBy == @event.IdentityAddress, CancellationToken.None);
```

Comment:
I think we don't have to read all the allocations. It's okay if we just read whether there is an allocation and return true or false from the database.

Same goes for the relationships above